pkg/k8shandler/certs.go

```diff
@@ -422,15 +426,40 @@ func (clusterRequest *ClusterLoggingRequest) populateCA() error {
 	return nil
 }
 
-func genCert(componentName string) (ret certificate, err error) {
+func (clusterRequest *ClusterLoggingRequest) incCertSerial() (*big.Int, error) {
+	secret, err := clusterRequest.GetSecret(caSecretName)
+	if err != nil {
+		return nil, err
+	}
+	cert, key, serialBytes, serial :=
+		func() ([]byte, []byte, []byte, *big.Int) {
+			cloCAMutex.Lock()
+			defer cloCAMutex.Unlock()
+			cloCA.serial.Add(cloCA.serial, bigOne)
+			serialCopy := big.NewInt(0)
+			serialCopy.Set(cloCA.serial)
+			return cloneByteSlice(cloCA.cert), cloneByteSlice(cloCA.key),
+				[]byte(cloCA.serial.Text(10)), serialCopy
+		}()
+	if err := clusterRequest.populateCASecret(secret, cert, key, serialBytes); err != nil {
+		return nil, err
+	}
+	return serial, nil
+}
+
+func (clusterRequest *ClusterLoggingRequest) genCert(componentName string) (ret certificate, err error) {
 	if ret.privKey, err = rsa.GenerateKey(rand.Reader, rsaKeyLength); err != nil {
 		return
 	}
 	pubKeySHA1 := sha1.Sum(x509.MarshalPKCS1PublicKey(&ret.privKey.PublicKey))
+	var serial *big.Int
+	if serial, err = clusterRequest.incCertSerial(); err != nil {
+		return
+	}
 	cloCAMutex.Lock()
 	defer cloCAMutex.Unlock()
```

Comment:
Can you isolate the shared state that needs locking in a smaller type to make it clear why we need this lock?
I noticed is the global `extensions` map is shared state, if that's the only thing, then you could replace it with a sync.Map and drop the lock.